### DISCUSSION TITLE
Unable to check reports flow in local

### DISCUSSION THREAD

**Discussion by kiranharidas187** [Sunbird-Ed/Community#11] [Bugs] [answered]
@sharathkashyap @luckynvdu 

While running the SunbirdEd portal locally, we are unable to test the report flows. All the report-related APIs are giving 403 FORBIDDEN errors. But all other APIs are working fine

**Solutions tried**

- Tried in both dev and staging environments
- Disabled whitelist API env variable to false, in both local and dev server env variable
- Tried getting the same token from the dev server with help of the DevOps team, and replace with local variables

**Sample curl from local**

curl 'http://localhost:3000/report/request/list/6204f813841a270008e83932_16953686-0e88-4482-b86f-b34bbfb79e80' \
  -H 'Accept: application/json, text/plain, */*' \
  -H 'Accept-Language: en-GB,en-US;q=0.9,en;q=0.8' \
  -H 'Connection: keep-alive' \
  -H 'Content-Type: application/json' \
  -H 'Cookie: connect.sid=s%3AIN-ycwVzNo9_w0OwBF2spHiTycylbhzv.d%2Bl%2FECE3InXMw5rcmZDn1pM%2BwQpohTumB17SekHITCs' \
  -H 'DNT: 1' \
  -H 'Referer: http://localhost:3000/program' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Site: same-origin' \
  -H 'User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/100.0.4896.75 Safari/537.36' \
  -H 'sec-ch-ua: " Not A;Brand";v="99", "Chromium";v="100", "Google Chrome";v="100"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "Linux"' \
  --compressed

Sample Response

{
    "id": "api.error",
    "ver": "1.0",
    "ts": "2022-05-26 04:03:07:112+0000",
    "params": {
        "resmsgid": "bfbd6680-dca8-11ec-866f-09115c12a8cf",
        "msgid": null,
        "status": "failed",
        "err": "FORBIDDEN_ERROR",
        "errmsg": "Forbidden: API WHITELIST Access is denied"
    },
    "responseCode": "FORBIDDEN",
    "result": {}
}

We are stuck with development for release-4.9.1 because of this

@vijiurs @5Amogh

**Comment by rajeevsathish** [comment, marked as answer]
@kiranharidas187 this route /report/request/list/:tag is whitelisted in portal backend with the following roles please check the user you are trying has the following role
      ORG_ADMIN,
    CONTENT_CREATOR,
    COURSE_MENTOR,
    PROGRAM_DESIGNER,
    PROGRAM_MANAGER
if not this whitelisting error will throw hope this helps.

**Reply by kiranharidas187** [reply]
@rajeevsathish yes account has required roles. the same account works in the deployed portal (in both staging and dev). Only in local its not working

**Reply by rajeevsathish** [reply]
@kiranharidas187  on local you can change the code so that you will not go through the whitelisting flow

**Reply by kiranharidas187** [reply]
@rajeevsathish Sorry didn't understand. Can you let me know what changes I need to do??  In env we changed whitelisting flag to false and even then same issue was happening

**Reply by kiranharidas187** [reply]
@rajeevsathish can you help with this??  We are stuck with this for 4.9.1  sprint completion

**Reply by kiranharidas187** [reply]
@rajeevsathish @sharathkashyap can someone help in this??

cc- @vijiurs

**Reply by rajeevsathish** [reply]
@kiranharidas187 you can de-activate the whitelisting api flow by making this variable as false

PORTAL_API_WHITELIST_CHECK 

if this is false  in the apiWhitelist.js file it will move to the else part in line number 76 
if you update any env value please check it has take effect and restart your server post changing and also you can Put break points and check the same

**Reply by sharathkashyap** [reply]
@kiranharidas187 can you confirm if the issue is resolved post RBAC changes

**Reply by kiranharidas187** [reply]
yes, it's working. Thanks, @sharathkashyap @rajeevsathish
